### PR TITLE
First-class build in Step

### LINKED ISSUE #200: First-class build
Instead of `run: build ...`, use `build: ...`

### PR DESCRIPTION
**Pull request by ehotinger**
**Purpose of the PR:**

First-classes `build` as a property of `Step`, see the changed yaml for usage references.

**Fixes #200**

### CODE REVIEW

File: builder/builder.go [hunk 1]
```diff
@@ -182,12 +182,12 @@ func (b *Builder) runStep(ctx context.Context, step *graph.Step) error {
...
 
-			step.Cmd = replacePositionalContext(step.Cmd, ".")
+			step.Build = replacePositionalContext(step.Build, ".")
 		}
 
 		args = b.getDockerRunArgs(volName, workDir, step)
-		args = append(args, "docker")
-		args = append(args, strings.Fields(step.Cmd)...)
+		args = append(args, "docker", "build")
+		args = append(args, strings.Fields(step.Build)...)
```
Comment (northtyphoon) [inline builder/builder.go]: >(args, strings.Fields(step.Build) [](start = 15, length = 33)

Are users already be able to to pass arbitrary arguments to build and run commands?

**Comment by northtyphoon** [comment]
> 			args = append(args, "--env", env)

--env --entrypoint don't apply to build, is it correct?

---
Refers to: builder/builder.go:194 in 1a38e87. [](commit_id = 1a38e87c0c452bdee5753e26dc5fd81a1a925cb7, deletion_comment = False)

File: builder/builder.go (hunk 1 above)
Comment (ehotinger) [reply builder/builder.go]: yes

**Comment by ehotinger** [comment]
@northtyphoon correct, those don't apply to `build`

**Pull request merged**